firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+KeyCommands.swift

```diff
@@ -55,11 +55,15 @@ extension BrowserViewController {
                                      method: .press,
                                      object: .keyCommand,
                                      extras: ["action": "reload"])
-
-        guard let tab = tabManager.selectedTab else { return }
-
-        if !contentContainer.hasHomepage {
-            tab.reload()
+        if isToolbarRefactorEnabled {
+            store.dispatch(GeneralBrowserAction(windowUUID: windowUUID,
+                                                actionType: GeneralBrowserActionType.reloadWebsite))
+        } else {
+            guard let tab = tabManager.selectedTab else { return }
```

Comment:
I think the guard for all of these cases should be before the feature flag check so we don't have to dispatch an action if there is no selected tab (and whatever other conditions the guard has).

Reply:
Ok true, I'm going to modify those